interview-questions.md

```diff
@@ -8,44 +8,45 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-Your answer:
+Your answer:  A Hash is a collection of unique keys and their values. Also called associative arrays, they are similar to Arrays, but where an Array uses integers as its index, a Hash allows you to use any object type. Hashes enumerate their values in the order that the corresponding keys were inserted. Hashes allow an alternate syntax for keys that are symbols. Hashes have a default value that is returned when accessing keys that do not exist in the hash. If no default is set nil is used.
 
-Researched answer:
+Researched answer:  The order in which you traverse a hash by either key or value may seem arbitrary and will generally not be in the insertion order. If you attempt to access a hash with a key that does not exist, the method will return nil. In Ruby, Hash is a set of keyt value pairs. With a hash, the indexing is done with the help of keys of any object type. In Hash, the order of returning keys and their value by various iterators is arbitrary and will generally not be in the insertion order. The default value of Hashes is nil. When a user tries to access the keys which do not exist in the hash, then the nil value is returned. 
```

Comment:
[Found the article you copied this from. ](https://www.tutorialspoint.com/ruby/ruby_hashes.htm#:~:text=The%20order%20in%20which%20you%20traverse%20a%20hash%20by%20either%20key%20or%20value%20may%20seem%20arbitrary%20and%20will%20generally%20not%20be%20in%20the%20insertion%20order.%20If%20you%20attempt%20to%20access%20a%20hash%20with%20a%20key%20that%20does%20not%20exist%2C%20the%20method%20will%20return%20nil.)

Reply:
understood